### PR TITLE
`xtask`: Simplify `fn utils::run_command`

### PR DESCRIPTION
**Pull request by Urhengulas**
The changes in `main.rs` are the adaptions to the function signature change.

### CODE REVIEW

File: xtask/src/utils.rs [hunk 1]
```diff
@@ -24,34 +24,30 @@ pub fn run_capturing_stdout(cmd: &mut Command) -> anyhow::Result<String> {
...
-    }
-
-    let cwd_s = if let Some(path_ref) = cwd {
-        cmd = cmd.current_dir(path_ref);
-        format!("CWD:{} ", path_ref)
+pub fn run_command(
+    program: &str,
+    args: &[&str],
+    cwd: Option<&str>,
+    envs: Vec<(&str, &str)>,
```
Comment (jonas-schievink) [inline xtask/src/utils.rs:31]: You shouldn't need to pass a `Vec` here, a slice should work fine (perhaps with a `.copied()` somewhere in this function), that should be a bit simpler

Comment (Urhengulas) [reply xtask/src/utils.rs:31]: I tried it with the `envs: &[(&str, &str)]` first, but this didn't work and the only solution I figured out is `Vec::from(envs)`. Doing the vector creation in the functions makes the api easier, but is a lie 😬

Comment (Urhengulas) [reply xtask/src/utils.rs:31]: With just cmd.envs(&[(&str, &str]);` I get:
```
error[E0271]: type mismatch resolving `<&[(&str, &str)] as IntoIterator>::Item == (_, _)`
  --> xtask/src/utils.rs:34:9
   |
34 |     cmd.envs(env);
   |         ^^^^ expected reference, found tuple
   |
   = note: expected reference `&(&str, &str)`
                  found tuple `(_, _)`
```

Comment (jonas-schievink) [reply xtask/src/utils.rs:31]: What happens if you pass `env.iter().copied()`?

Comment (Urhengulas) [reply xtask/src/utils.rs:31]: Noooooo, it works 😭 ... I mean yay! 🙂

**Pull request merged**